errors/TypesDoNotUnify.md

```diff
@@ -33,6 +33,9 @@ In the example above, the types `String` and `Int` can never be made equal, henc
 - Look carefully at the error, especially the information at the end. Usually, it will help to narrow down the offending expression. For example, in the error message above, we are told that the error occurred "while checking that expression `"test"` has type `Int`".
 
 ## Notes
+### Arrays
+
+Unlike in Javascript, all elements of a PureScript array must be the same type. For example, ```[true, false]``` or ```[1, 2, 3]``` work fine but ```[1, false]``` does not.
```

Comment:
@LiamGoodacre Thanks!

@paf31 I don't think I know Haskell/Purescript internals enough to adequately illustrate the use of unification here or why it is necessary.